compliance/azure/azure_untagged_resources/untagged_resources.pt

```diff
@@ -177,65 +176,106 @@ script "js_parse_tag_support", type: "javascript" do
+  _.each(ds_azure_resources, function(resource){
+
+    //Create a field for Missing tags
+    resource["missingTagKeys"] = []
+    resource["missingTagValues"] = []
+
+    //For each Tag in Param Tag List
+    _.each(param_tags_to_check, function(param){
+      // Find Param Tag Key in Resource Tag Array
+      current_tag = _.find(resource.tags, function(tag){ return tag.tagKey == param })
+      // If Tag Key does not exist for Resource, create Missing Tag field and set to True
+      if ( current_tag == undefined || current_tag == null) {
+        resource["isMissingTagKey"] = true
+        resource.missingTagKeys.push(param)
       }
-      for(var j=0; j < param_tag_key_lower.length; j++){
-        if(_.isEmpty(resource_tags) || (resource_tags.toLowerCase().indexOf(param_tag_key_lower[j].toLowerCase()))==-1){
-          isResMissedTag=true;
-          all_missed_tag=all_missed_tag+" "+param_tag_key_lower[j];
-        }else{
-          all_include_tag=all_include_tag+" "+param_tag_key_lower[j];
+      // Otherwise, if Tag Key does exist...
+      else {
+        // and if Tag Value is empty, create Missing Tag field and set to True
+        if ( current_tag.tagValue == "" || current_tag.tagValue == null ) {
+          resource["isMissingTagValue"] = true
+          resource.missingTagValues.push(param)
         }
       }
-    if((isResMissedTag)&&(tag_support!=='FALSE')){
+    })
+  })
+
+  //Filter out all tags where isMissingTag != true
+  filtered_resources = _.filter(ds_azure_resources, function(resource){ return resource.isMissingTagKey == true || resource.isMissingTagValue == true })
+
+
+  _.each(filtered_resources, function(resource){
+
+    // Create string for existing Tags
+    var existing_tags = ""
+    _.each(resource.tags, function(tag){
+      if ( existing_tags == "" ) { existing_tags += tag.tagKey + "=" + tag.tagValue }
+      else { existing_tags += ", " + tag.tagKey + "=" + tag.tagValue }
+    })
+
+    var tag_support=ds_parsed_tag_support[resource["type"].toLowerCase()];
+
+    if((resource["isMissingTagKey"] || resource["isMissingTagValue"])&&(tag_support!=='FALSE')){
       result.push({
-        subscriptionId: ar["subscriptionId"],
-        subscriptionName : ar["subscriptionName"],
-        id: ar["id"],
-        region: ar["region"],
-        name: ar["name"],
-        resource_group: resource_group_name,
-        namespace: namespace,
-        resource_type: ar["type"],
-        all_missed_tag: all_missed_tag,
-        all_include_tag: resource_tags,
-        all_tags: all_tags,
+        subscriptionId: resource["subscriptionId"],
+        subscriptionName : resource["subscriptionName"],
+        id: resource["id"],
+        region: resource["region"],
+        name: resource["name"],
+        resource_group: resource["resourceGroup"],
+        namespace: resource["namespace"],
+        resource_type: resource["type"],
+        all_missed_tag_keys: resource["missingTagKeys"]
```

Comment:
Lines 266 and 267: think we should consistent in the naming convention of these fields across both policies - would suggest `missingTagKeys `and `missingTagValues `respectively